CBCCasper.thy

```diff
@@ -69,7 +38,32 @@ fun justification :: "message \<Rightarrow> state"
   where
     "justification (Message (_, _, s)) = set s"
 
-(* Definition 2.7 *)
+record params =
+  V :: "validator set"
+  W :: weight
+  t :: threshold
+  C :: "consensus_value set"
+  \<epsilon> :: "message set \<Rightarrow> consensus_value set"
+
+record Protocol =
+  parameters :: params
+  \<Sigma> :: "state set"
+  M :: "message set"
+
+locale Protocol =
+  fixes protocol :: "'p Protocol_scheme" (structure)
+  and params
```

Comment:
Instead of fixing `params` here, how about fixing `V`, `W`, `t`, `C` and `ε` so that we can use them like just `V` (not `V params`) later?

Reply:
I did this because I wanted to define `ConstructiveProtocol` without changing these parameters. Anyways I'm going to remove `ConstructiveProtocol` so I will do so.